oshi-core/src/main/java/oshi/hardware/platform/unix/openbsd/OpenBsdHWDiskStore.java

```diff
@@ -62,6 +57,68 @@ private OpenBsdHWDiskStore(String name, String model, String serial, long size)
         super(name, model, serial, size);
     }
 
+    /**
+     * Gets the disks on this machine.
+     *
+     * @return an {@code UnmodifiableList} of {@link HWDiskStore} objects representing the disks
+     */
+    public static List<HWDiskStore> getDisks() {
+        List<OpenBsdHWDiskStore> diskList = new ArrayList<>();
+
+        // Get list of disks from sysctl
+        // hw.disknames=sd0:2cf69345d371cd82,cd0:,sd1:
+        String[] devices = OpenBsdSysctlUtil.sysctl("hw.disknames", "").split(",");
+        OpenBsdHWDiskStore store;
+        for (String device : devices) {
+            // get partitions using disklabel command (requires root)
+            Pair<Map<String, String>, List<HWPartition>> diskdata = Disklabel
+                .getDiskParams((device.split(":")[0]));
+            if (diskdata.getA().isEmpty()) {
+                // something must have gone wrong, probably no elevated/root permission
+                store = new OpenBsdHWDiskStore((device.split(":")[0]), Constants.UNKNOWN, Constants.UNKNOWN, 0L);
+            } else {
+                store = new OpenBsdHWDiskStore(
+                    diskdata.getA().get("label"),
+                    diskdata.getA().get("label"),
+                    diskdata.getA().get("duid"),
+                    ParseUtil.parseLongOrDefault(diskdata.getA().get("bytes/sector"), 0) * ParseUtil
```

Comment:
Something just doesn't seem right using a Pair and then a map with String Values that have to be parsed and then later multiplied together.   Could this not all be a single Quintet or even just a class with sensible names and proper variable typing internally?  Alternately, if the "C" partition is the "whole disk" could you not just send back that partition separately with all these values populated so you'd have a `Pair<HWPartition, List<HWPartition>` where `getA()` is the whole-disk stuff and `getB()` is everything else?